### PR TITLE
feat: should show global backup submenu in persona setting menu

### PR DESCRIPTION
**Pull request by NikoManiac**
1. Should show global backup submenu in persona setting menu
2. Persona tab bar add restore button
3. Show all twitter profile that had connected
4. Should open backup dialog when goto setting from persona global backup button

### CODE REVIEW

File: packages/shared/src/utils/formatter.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export function formatFingerprint(fingerprint: string, size: number = 0) {
+    if (size === 0) return fingerprint
+    return `${fingerprint.substr(0, 2 + size)}...${fingerprint.substr(-size)}`
```
Comment (septs) [inline packages/shared/src/utils/formatter.ts:3]: why not using `String.prototype.slice`?

Comment (NikoManiac) [reply packages/shared/src/utils/formatter.ts:3]: In this scenario, both are ok.

**Review by yanzhihong23 — DISMISSED** [review]

File: packages/dashboard/src/pages/Personas/components/DisconnectProfileDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+interface DisconnectProfileDialogProps {
+    open: boolean
+    networkIdentifier: string
+    onClose(): void
+    onDisconnect: (identifier: ProfileIdentifier) => void
+    profileIdentifiers: ProfileIdentifier[]
+}
+
+enum steps {
+    selection = 0,
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/DisconnectProfileDialog/index.tsx:26]: Don't use `0`.

File: packages/dashboard/src/pages/Personas/components/DisconnectProfileDialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const [currentStep, setCurrentStep] = useState<steps>(steps.selection)
+    const [profileIdentifier, setProfileIdentifier] = useState<ProfileIdentifier | null>()
+
+    return (
+        <MaskDialog open={open} title={t.personas_disconnect()} onClose={onClose}>
+            <DialogContent sx={{ width: 440 }}>
+                {currentStep === steps.selection && (
+                    <Stack minHeight="100px" justifyContent="center">
+                        {profileIdentifiers.map((x) => (
+                            <Stack direction="row" alignItems="center" justifyContent="space-between" mb={1}>
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/DisconnectProfileDialog/index.tsx:49]: The `key` is lost.

**Review by yanzhihong23 — DISMISSED** [review]

File: packages/dashboard/src/pages/Personas/components/PersonaDrawer/index.tsx [hunk 4]
```diff
@@ -87,33 +82,50 @@ export const PersonaDrawerUI = memo<PersonaDrawerUIProps>(
...
-                    <Button onClick={() => setShowAddPersonaCard(true)}>{t.personas_add_persona()}</Button>
-                    <Button color="warning">{t.personas_back_up()}</Button>
-                </Box>
+                classes={{ paper: classes.paper }}>
+                <Stack justifyContent="space-between" gap={2} height="100%" maxHeight="100%">
+                    <Box overflow="auto">
+                        {personas.map((item) => {
+                            const { identifier, nickname, linkedProfiles } = item
+                            return (
+                                <Box mb={2.5}>
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/PersonaDrawer/index.tsx:91]: The `key` is lost.

**Review by guanbinrui — APPROVED** [review]

File: packages/maskbook/src/social-network-adaptor/minds.com/shared.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export const mindsShared: SocialNetwork.Shared & SocialNetwork.Base = {
     ...mindsBase,
     utils: {
         getHomePage: () => 'https://www.minds.com',
+        getProfilePage: () => 'https://www.minds.com',
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/minds.com/shared.ts:15]: This looks wrong. Remove this plz.

File: packages/maskbook/src/social-network-adaptor/instagram.com/shared.ts [hunk 6]
```diff
@@ -7,6 +7,7 @@ export const instagramShared: SocialNetwork.Shared & SocialNetwork.Base = {
     ...instagramBase,
     utils: {
         getHomePage: () => 'https://www.instagram.com/',
+        getProfilePage: () => 'https://www.instagram.com/',
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/instagram.com/shared.ts:10]: This looks wrong. Remove this plz.

File: packages/maskbook/src/social-network-adaptor/facebook.com/shared.ts [hunk 7]
```diff
@@ -14,6 +14,7 @@ export const facebookShared: SocialNetwork.Shared & SocialNetwork.Base = {
     ...facebookBase,
     utils: {
         getHomePage: () => 'https://www.facebook.com',
+        getProfilePage: (userId: string) => 'https://www.facebook.com',
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/facebook.com/shared.ts:17]: This looks wrong. Remove this plz.

File: packages/maskbook/src/extension/background-script/SocialNetworkService.ts [hunk 8]
```diff
@@ -29,3 +29,14 @@ export async function connectSocialNetwork(identifier: PersonaIdentifier, networ
     await delay(100)
     home && browser.tabs.create({ active: true, url: home })
 }
+
+export async function openProfilePage(network: string, userId?: string) {
+    const ui = await loadSocialNetworkUI(network)
+    const profile = ui.utils.getProfilePage?.(userId)
+    if (!Flags.no_web_extension_dynamic_permission_request) {
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/background-script/SocialNetworkService.ts:36]: Why? We can safely assume we already have the permission.

**Pull request merged**